src/neural-network.js

```diff
@@ -783,8 +805,9 @@ export default class NeuralNetwork {
    * @param json
    * @returns {NeuralNetwork}
    */
-  fromJSON (json) {
-    this.sizes = json.sizes
+  fromJSON (userJSON) {
+    const json = deepClone(userJSON);
```

Comment:
any reason we can't do something like `JSON.parse(JSON.stringify(`?

Reply:
bump

Reply:
I mean, we could...but that just _seems_ so hacky to me. Plus that benchmark test I ran seems like we could use the `deepClone` in spots where we need more performance than `JSON.parse(JSON.stringify(...))`

Reply:
You are right, it does seem hacky, I guess what I'd want to address is I'd like like not to have to support code we don't have to, is there a lib that gives us this, or an internal practice that node provides, rather than doing it ourselves?

Reply:
https://lodash.com/docs/4.17.4#cloneDeep